templates/addons/env/opensearch.yml

```diff
@@ -193,35 +192,32 @@ Resources:
         - Key: 'Copilot-Environment'
           Value: !Sub ${Env}
 
+  # An intermediate SSM parameter just to store the URL. This is required because attempting use a larger nested Fn::If / Fn::ToJsonString / !Sub / !GetAtt structure
+  # leads to unhelpful "ResourceNotReady: failed waiting for successful resource state: Nested change set " error.
   {{ service.prefix }}OpenSearchEndpointAddressParam:
     Type: AWS::SSM::Parameter
-    Condition: DisableHA
     Properties:
-      Name: !Sub "/copilot/${App}/${Env}/secrets/{{ service.secret_name }}"
       Type: String
-      Value:
-        Fn::ToJsonString:
-          url: !Sub
-            - "https://${url}"
-            - url: !GetAtt {{ service.prefix }}OpenSearchDomain.DomainEndpoint
-          username:
-            !Join [ "",  [ '{% raw %}{{resolve:secretsmanager:{% endraw %}', !Ref {{ service.prefix }}OpenSearchSecret, "{% raw %}:SecretString:username}}{% endraw %}" ]]
-          password:
-            !Join [ "",  [ '{% raw %}{{resolve:secretsmanager:{% endraw %}', !Ref {{ service.prefix }}OpenSearchSecret, "{% raw %}:SecretString:password}}{% endraw %}" ]]
+      Value: !If
+        - EnableHA
+        - !Sub
+          - "https://${url}"
+          - url: !GetAtt dwOpensearchOpenSearchDomainHA.DomainEndpoint
+        - !Sub
+          - "https://${url}"
+          - url: !GetAtt dwOpensearchOpenSearchDomain.DomainEndpoint
```

Comment:
`dwOpensearchOpenSearchDomain` -> Looks like this is specific to `dw`(?).

Reply:
Might need replacing with the `service.prefix`.

Reply:
Yep, doh.

Reply:
Resolved in https://github.com/uktrade/copilot-tools/pull/82/commits/fc0d2eb458b574a2d80fe900d85f0a9f03c2f5a7.